src/core/layertree/qgslayertreemodel.cpp

```diff
@@ -316,6 +320,14 @@ QVariant QgsLayerTreeModel::data( const QModelIndex &index, int role ) const
 
         parts << "<i>" + source.toHtmlEscaped() + "</i>";
 
+        QgsLayerTreeLayer *nodeLayer = QgsLayerTree::toLayer( node );
+        const bool showFeatureCount = nodeLayer->customProperty( QStringLiteral( "showFeatureCount" ), 0 ).toBool();
+        const bool estimatedCount = QgsDataSourceUri( layer->dataProvider()->dataSourceUri() ).useEstimatedMetadata();
+        if ( showFeatureCount && estimatedCount )
+        {
+          parts << "<b>Feature Count is estimated</b> : Please consider keeping database statistics up to date";
```

Comment:
This sentence should inform the user that "the feature count is determined by the database statistics".
"Please consider ..." sounds like a suggestion, but actually it's well possible the statistics are up to date already.

Reply:
Agreed